app/util/Legend.js

```diff
@@ -119,6 +124,20 @@ Ext.define('CpsiMapview.util.Legend', {
      */
     getSldFileFromWmsStyle: function (wmsStyle, wmsLayers) {
         return wmsLayers + '_' + wmsStyle.replace(' ', '_') + '.xml';
-    }
+    },
+
+    /**
+     * Removes any duplicate layer name from the given LAYERS parameter value.
+     *
+     * @param  {String} layers WMS LAYERS parameter value, e.g. (layer1,layer2)
+     * @return {String}        WMS LAYERS parameter without duplicates
+     */
+    getUniqueLayersParam: function (layers) {
+        var layerList = Ext.isArray(layers) ? layers : layers.split(',');
+        // remove any duplicate layer names
+        var reducedLayerList = Ext.Array.unique(layerList);
+
+        return reducedLayerList.join(',');
+    },
```

Comment:
Extra comma here should be removed?

Reply:
I don't think so. `join(',')` handles this and avoids a trailing comma. See [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join). Or do I oversee something?

Reply:
Sorry I meant line 141. This used to break things on IE7 :)

Reply:
Oh now I see ... yes the good old IE 7 times :laughing: I guess the JS builder takes care of this, but to be save I'll remove this.